hivemq-edge/src/frontend/src/extensions/datahub/components/helpers/MessageInterpolationTextArea.tsx

```diff
@@ -0,0 +1,79 @@
+import { ChangeEvent, FC, FocusEvent, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { labelValue, WidgetProps } from '@rjsf/utils'
+import { getChakra } from '@rjsf/chakra-ui/lib/utils'
+import { Button, ButtonGroup, FormControl, FormLabel, Textarea, VStack } from '@chakra-ui/react'
+
+export const MessageInterpolationTextArea = (props: WidgetProps) => {
+  const { t } = useTranslation('datahub')
+  const chakraProps = getChakra({ uiSchema: props.uiSchema })
+
+  const [cur, setCur] = useState<[number, number] | undefined>(undefined)
+
+  const _onChange = ({ target: { value, selectionStart, selectionEnd } }: ChangeEvent<HTMLTextAreaElement>) => {
+    props.onChange(value === '' ? props.options.emptyValue : value)
+    setCur([selectionStart, selectionEnd])
+  }
+  const _onBlur = ({ target: { value, selectionStart, selectionEnd } }: FocusEvent<HTMLTextAreaElement>) => {
+    // TODO[NVL] the state change impacts on the blur
+    setCur([selectionStart, selectionEnd])
+    props.onBlur(props.id, value)
+  }
+  const _onFocus = ({ target: { value } }: FocusEvent<HTMLTextAreaElement>) => {
+    props.onFocus(props.id, value)
+  }
```

Comment:
question(non-blocking): is the private decoration needed? I first was very confused why it's the manner. Until I realised it's most likely done because of the name conflict with the given component `props`. Yet since we use `props.onBlur` I think we can also rename the custom handlers.

```suggestion
  const onChange = ({ target: { value, selectionStart, selectionEnd } }: ChangeEvent<HTMLTextAreaElement>) => {
    props.onChange(value === '' ? props.options.emptyValue : value)
    setCur([selectionStart, selectionEnd])
  }
  const onBlur = ({ target: { value, selectionStart, selectionEnd } }: FocusEvent<HTMLTextAreaElement>) => {
    // TODO[NVL] the state change impacts on the blur
    setCur([selectionStart, selectionEnd])
    props.onBlur(props.id, value)
  }
  const onFocus = ({ target: { value } }: FocusEvent<HTMLTextAreaElement>) => {
    props.onFocus(props.id, value)
  }
```